setuptools/command/build_ext.py

```diff
@@ -402,7 +402,7 @@ def link_shared_object(
         library_dirs=None,
         runtime_library_dirs=None,
         export_symbols=None,
-        debug=False,
+        debug: bool = False,
```

Comment:
This is a funny one...

Does this mean that when we define a class attribute like `inplace = False` the inferred type is already `bool`, but if we have a function argument with `debug=False` the inferred type is `Literal[False]`?

That sounds like the type-checker is very inconsistent...

Reply:
Both are inferred as `bool`.

Mypy only checks the body of typed functions, and a function isn't considered typed until all its params are annotated (*or* a return type is explicitly added for functions with no parameters).

Pyright has the `reportMissingParameterType` check: https://github.com/microsoft/pyright/blob/main/docs/configuration.md

https://docs.astral.sh/ruff/rules/missing-return-type-undocumented-public-function/ won't trigger for a public function with no annotation even if all params have default values.

Until all `link_shared_object` params are annotated, it won't change anything for mypy. So if it feels inconsistent to you, I can remove that annotation for now.

Reply:
Ah, ok I see, it is not about the inferred type, it is about when a function is considered typed.

I don't have any problems in leaving this as you suggested in the PR changes.